strawberryfields/backends/fockbackend/ops.py

```diff
@@ -500,3 +498,68 @@ def hermiteVals(q_mag, num_bins, m_omega_over_hbar, trunc):
         Hvals[i] = 2 * x * Hvals[i - 1] - 2 * (i - 1) * Hvals[i - 2]
 
     return q_tensor, Hvals
+
+
+def gkp_displacements(t, k, epsilon):
+    """
+    Helper function to generate the displacements parameters associated with the teeth of
+    GKP computational basis state k.
+    """
+    return np.sqrt(0.5 * np.pi) * (2 * t + k) / np.cosh(epsilon)
+
+
+def gkp_coeffs(t, k, epsilon):
+    """
+    Helper function to generate the coefficient parameters associated with the teeth of
+    GKP computational basis state k.
+    """
+    return np.exp(-0.5 * np.pi * np.tanh(epsilon) * (k + 2 * t) ** 2)
+
+
+@functools.lru_cache()
+def squaregkpBasisState(i, epsilon, ampl_cutoff, cutoff):
+    """
+    Generate the Fock expansion of a computational GKP basis state.
+
+    Args:
+        i (int): a computational basis state label, can be either 0 or 1
+        epsilon (float): finite energy parameter of the state
+        ampl_cutoff (float): this determines how many terms to keep in the Hilbert space expansion
+        cutoff (int): Fock space truncation
+
+    Returns
+        (array): the expansion of the ith computational basis state in the Fock basis
+
+    """
+    z_max = int(np.ceil(np.sqrt(-0.25 / np.pi * np.log(ampl_cutoff) / np.tanh(epsilon))))
+    coeffs = [gkp_coeffs(t, i, epsilon) for t in range(-z_max, z_max + 1)]
+    r = -0.5 * np.log(np.tanh(epsilon))
+    alphas = [gkp_displacements(t, i, epsilon) for t in range(-z_max, z_max + 1)]
```

Comment:
Just a thought: we're iterating `range(-z_max, z_max + 1)` in two list comprehensions, would have a `for` loop and appending to two separate lists be considerably more efficient? 🤔

Reply:
yes, it would be maybe twice as fast but not much more. I have to admit that I like better to write list comprehensions that to append things into a lits.